test/parallel/test-string-decoder.js

```diff
@@ -97,6 +97,21 @@ assert.strictEqual(decoder.lastTotal, 3);
 
 assert.strictEqual(decoder.end(), '\ufffd');
 
+// TypedArray with Int8Array test
+decoder = new StringDecoder('utf8');
+assert.strictEqual(decoder.write(new Int8Array([1, 2])), '\u0001\u0002');
```

Comment:
I'd prefer this test use [getArrayBufferViews()](https://github.com/nodejs/node/blob/master/test/common/README.md#getarraybufferviewsbuf) to cover all possible TypedArray types and DataView. An example usage of this function by @SirR4T is

https://github.com/nodejs/node/blob/be5e3964b316da013d02ee743420e5fd9cb50c76/test/parallel/test-fs-write-file-typedarrays.js#L24-L31